### ISSUE TITLE
Is jsonifier slower than aeson on real-world data?

### ISSUE THREAD

**Issue by mitchellwrosen** [nikita-volkov/jsonifier#5]
In [this comment](https://github.com/mitchellwrosen/text-ansi/discussions/5#discussioncomment-4584601), @KtorZ reports that `jsonifier` is slower than `aeson`.

Let's get to the bottom of this! @KtorZ, could you share  any benchmarks you have that demonstrate this, or maybe a link to a discussion from some other forum?

**Comment by KtorZ** [comment]
To give some context, this comment mostly stemmed from empirical measurement and benchmarks using profiling on this project: https://github.com/CardanoSolutions/ogmios -- which has heavy needs on JSON-serialization (hundred of megabytes of data per second). 

I originally planned to give `jsonifier` a try after seeing some blog post / reddit post about it, and as recorded [here](https://github.com/CardanoSolutions/ogmios/blob/master/architectural-decisions/superseded/005-server_use-jsonifier.md). I ended up abandoning this decision in favor of aeson's encoding API (see [006](https://github.com/CardanoSolutions/ogmios/blob/master/architectural-decisions/accepted/006-server_pass-explicit-json-codecs.md)) which turned out to be more interoperable and actually faster than `jsonifier` on the profiling. 

What is also absolutely certain is that the default API of aeson (constructing Haskell `Value` and then serializing) is an order of magnitude slower than encoding JSON directly. 

Note also that, this happened 2 years ago. I don't know what's the state of the library today.

**Comment by mitchellwrosen** [comment]
Excellent, thanks for the context.

**Comment by nikita-volkov** [comment]
Hey guys! Thanks for the report. Is it possible to narrow this down to a benchmark or a set of benchmarks that highlights the difference?

**Comment by KtorZ** [comment]
Just to be clear from my end: I am not particularly interested in this conversation / outcome. I've got other matters to tackle and I was merely providing a bit of context.

Best of luck 🍀

**Comment by nikita-volkov** [comment]
@mitchellwrosen So where do we go from here? The specified links only provide vague claims and no way to reproduce. I'm really interested in getting to the bottom of this.

**Comment by mitchellwrosen** [comment]
Indeed; I had considered trying to put together a benchmark. I'm sure enormous gobs of "real world" JSON aren't hard to generate or get a hold of.

However, life took over, so I haven't gotten around to doing anything yet.

Feel free to close this issue if you'd like - I may come through with some benchmarks at a later date and re-open if no one else has investigated by then.
